ads/inabox/inabox-messaging-host.js

```diff
@@ -60,23 +106,78 @@ export class InaboxMessagingHost {
       return false;
     }
 
-    if (request.type == MessageType.SEND_POSITIONS) {
-      // To prevent double tracking for the same requester.
-      if (this.registeredIframeSentinels_[request.sentinel]) {
-        return false;
-      }
-      this.registeredIframeSentinels_[request.sentinel] = true;
-      this.positionObserver_.observe(iframe, data => {
-        dev().fine(TAG, `Sent position data to [${request.sentinel}]`, data);
-        message.source./*OK*/postMessage(
-            serializeMessage(MessageType.POSITION, request.sentinel, data),
-            message.origin);
-      });
-      return true;
-    } else {
+    if (!this.msgObservable_.fire(request.type, this,
+        [iframe, request, message.source, message.origin])) {
       dev().warn(TAG, 'Unprocessed AMP message:', message);
       return false;
     }
+
+    return true;
+  }
+
+  /**
+   * @param {!HTMLIFrameElement} iframe
+   * @param {!Object} request
+   * @param {!Window} source
+   * @param {string} origin
+   * @return {boolean}
+   */
+  handleSendPositions_(iframe, request, source, origin) {
+    // To prevent double tracking for the same requester.
+    if (this.registeredIframeSentinels_[request.sentinel]) {
+      return false;
+    }
+    this.registeredIframeSentinels_[request.sentinel] = true;
+    this.positionObserver_.observe(iframe, data => {
+      dev().fine(TAG, `Sent position data to [${request.sentinel}]`, data);
+      source./*OK*/postMessage(
+          serializeMessage(MessageType.POSITION, request.sentinel, data),
+          origin);
+    });
+    return true;
+  }
+
+  /**
+   * @param {!HTMLIFrameElement} iframe
+   * @param {!Object} request
+   * @param {!Window} source
+   * @param {string} origin
+   * @return {boolean}
+   */
+  // TODO(alanorozco):
+  // 1. Reject request if frame is out of focus
+  // 2. Disable zoom and scroll on parent doc
+  handleEnterFullOverlay_(iframe, request, source, origin) {
+    expandFrame(this.win_, iframe, () => {
+      source./*OK*/postMessage(
+          serializeMessage(
+              MessageType.FULL_OVERLAY_FRAME_RESPONSE,
+              request.sentinel,
+              {content: {accept: true}}),
```

Comment:
I nested the content since `serializeMessage` adds top-level members and that's a little weird :/

Renaming to top-level `success` is still alright I guess. It's a simple API. Changed.